### ISSUE TITLE
Muitos recursos tornam alguns inacessíveis

### ISSUE THREAD

**Issue by xvitxr7** [Os-Derivados/college-management#42]
O usuário é incapaz de selecionar certos recursos em contextos devido ao modo que a entrada é controlada. Ao tentar acessar ``[10] Editar Matricula`` digitando o número _10_, o recurso ``[1] Visualizar`` é escolhido no lugar, pois o número _1_ é digitado primeiro.
![image](https://github.com/user-attachments/assets/dd78807f-dfa7-4e03-b733-c4f481d6834b)

Não é necessariamente um bug, mas é algo para se notar. Avisem-me se issues pequenas como essa não são necessárias.

**Comment by trsaints** [comment]
Acredito que de fato esteja incoerente com a lógica inicial. Sugiro que o que podemos fazer é paginar as opções de escolha.

**Comment by xvitxr7** [comment]
Concordo. Eu já tinha pensado num modo mais interativo, como um menu com seleção usando um cursor no terminal.

**Comment by xvitxr7** [comment]
> Daria para fazer uma TUI (Terminal User Interface) e usuários as setas do teclado para percorrer entre as opções.

Eu também pensei nisso e até que seria fácil implementar como uma View, e já que estamos falando disso, também seria interessante outras Views parecidas, como uma idêntica ao que você descreveu, porém de múltipla escolha, substituindo algumas funções de pesquisa.
Não sei se deveríamos focar nessas _features_ visto que ainda temos algumas porções cruciais do projeto a serem desenvolvidas ainda. Eu, particularmente, não vejo porquê não.

**Comment by xvitxr7** [comment]
> > Eu também pensei nisso e até que seria fácil implementar como uma View, e já que estamos falando disso, também seria interessante outras Views parecidas, como uma idêntica ao que você descreveu, porém de múltipla escolha, substituindo algumas funções de pesquisa.
> > Não sei se deveríamos focar nessas _features_ visto que ainda temos algumas porções cruciais do projeto a serem desenvolvidas ainda. Eu, particularmente, não vejo porquê não.
> 
> Acho que se optarmos por atualizar isso agora, seria melhor no futuro (já que não teríamos que refazer o trabalho), poderíamos adotar uma espécie de MVC (Modelo, View e Controller). Temos uma camada de View já, só teríamos que refazer algumas coisas, se optarmos por fazer a navegação seria mais simples e intuitiva para o usuário e resolveria alguns problemas de input que temos com apenas um número. Mas entendo que ainda tem outras questões que são importantes no projeto, mas poderíamos começar com as telas e funcionalidades que já estão concluídas.
> 
> Isso já é algo que eu penso em fazer em um projeto meu, acho que não seria difícil de fazer. Caso o pessoal achar que vale, eu estaria super disposto em ajudar

Seu ponto é válido, porém acho que precisemos discutir melhor sobre; ver se encontramos um consenso. Caso a ideia venha a ser aprovada, eu também estou disposto a ajudar.

**Comment by xvitxr7** [comment]
Estes problemas estão sendo trabalhados em #21.
